msdk-io/msdk-io-mzml2/src/test/java/io/github/msdk/io/mzml2/MzMLFileWriterTest.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * (C) Copyright 2015-2017 by MSDK Development Team
+ *
+ * This software is dual-licensed under either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1 as published by the Free
+ * Software Foundation
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by the Eclipse Foundation.
+ */
+
+package io.github.msdk.io.mzml2;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import io.github.msdk.MSDKException;
+import io.github.msdk.datamodel.chromatograms.Chromatogram;
+import io.github.msdk.datamodel.chromatograms.ChromatogramType;
+import io.github.msdk.datamodel.msspectra.MsSpectrumType;
+import io.github.msdk.datamodel.rawdata.ActivationType;
+import io.github.msdk.datamodel.rawdata.MsScan;
+import io.github.msdk.datamodel.rawdata.PolarityType;
+import io.github.msdk.datamodel.rawdata.RawDataFile;
+import io.github.msdk.io.mzml2.data.MzMLCompressionType;
+import io.github.msdk.util.MsSpectrumUtil;
+
+public class MzMLFileWriterTest {
+
+  private Path getResourcePath(String resource) throws MSDKException {
+    final URL url = MzMLFileParserTest.class.getClassLoader().getResource(resource);
+    try {
+      return Paths.get(url.toURI()).toAbsolutePath();
+    } catch (URISyntaxException e) {
+      throw new MSDKException(e);
+    }
+  }
+
+  @Test
+  public void test5peptideFT() throws MSDKException, IOException {
+
+    float intensityBuffer[];
+
+    // Import the file
+    String file = "5peptideFT.mzML";
+    File inputFile = getResourcePath(file).toFile();
+    Assert.assertTrue(inputFile.canRead());
+    MzMLFileParser parser = new MzMLFileParser(inputFile);
+    RawDataFile rawFile = parser.execute();
+    Assert.assertNotNull(rawFile);
+    Assert.assertEquals(1.0, parser.getFinishedPercentage(), 0.0001);
+
+    // Export the file to a new mzML
+    File tempFile = File.createTempFile("msdk", ".mzML");
+    MzMLFileWriter exporter = new MzMLFileWriter(rawFile, tempFile,
```

Comment:
Since the current data model stores intensity and retention time values as float variables (which do cannot be MSNumpress compressed), I though of having two different compression.
Otherwise, we will have two options which have their own disadvantages -

- Convert them to double every time before compression, we will have to store additional `double[]`s
- Remove support for MSNumpress compression itself